pkgs/development/compilers/julia/1.0.nix

```diff
@@ -1,211 +1,154 @@
-{ lib, stdenv, fetchpatch, fetchurl, fetchzip
-# build tools
-, gfortran, m4, makeWrapper, patchelf, perl, which, python2
-, cmake
-# libjulia dependencies
-, libunwind, readline, utf8proc, zlib
-# standard library dependencies
-, curl, fftwSinglePrec, fftw, gmp, libgit2, mpfr, openlibm, openspecfun, pcre2
-# linear algebra
-, blas, lapack, arpack
-# Darwin frameworks
-, CoreServices, ApplicationServices
+{
+fetchzip, fetchFromGitHub, lib, pkgs, stdenv
+# Build tools
+, cmake, perl, python3, which
+# Dependencies
+, arpack, curl, blas, gmp, lapack, libssh2, libunwind, mpfr, pcre2
+# Testing
+, openssl
+# Darwin
+, ApplicationServices, CoreServices
 }:
 
+# Ensure dependencies from Nixpkgs are equal or greater to vendored versions.
+assert lib.versionAtLeast (lib.strings.getVersion gmp) "6.1.2";
+assert lib.versionAtLeast arpack.version "3.3.0";
+assert lib.versionAtLeast curl.version "7.56.0";
+assert lib.versionAtLeast libssh2.version "1.8.2";
+assert lib.versionAtLeast libunwind.version "1.1";
+assert lib.versionAtLeast mpfr.version "4.0.2";
+assert lib.versionAtLeast pcre2.version "10.30";
```

Comment:
Does the configure script not detect this?

Reply:
Excellent question, it is actually the case that there is no `configure` script. Only pure GNU Make. '^^ From what I can tell with a few `grep` calls, the only version check present is for the version of macOS.